src/linker/Linker/MessageCode.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Mono.Linker {
+	public enum MessageCode {
+		IL1000, // Error
```

Comment:
Why do we need both code and category if code range included category data?

Reply:
The message codes should actually represent specific errors/warnings. Having the numbering scheme to differentiate between errors and warnings is helpful though. Basically once we start using this system each failure should get its own code (and it should fail within a certain range, that's all).

Reply:
I agree that having specific ranges for different kind of issues and that's why I am not sure we need to duplicate the information. If we settle down to for example

0 - 1000 error
1001 - 10000 warning
10001 - 490000 for the rest

50000 - * for custom steps

then we should either pass the code in as int with enum flag or different method name where we could verify it or build this massive enum list but why to do both?

Reply:
Good point - that makes sense. With one glitch: If custom steps go to 50000+ those will also need to be able to to differentiate error/warning/info...

Reply:
Right so that's why I think
```c#
public MSBuildMessageContainer (int code, MessageCategory category, string message, string subcategory = null, MessageOrigin origin = null)
{
// chech if code matches category range
}
```

might work better